test/vvenc_unit_test/vvenc_unit_test.cpp

```diff
@@ -276,6 +276,66 @@ static bool test_IntraPred()
 #endif // ENABLE_SIMD_OPT_INTRAPRED
 
 #if ENABLE_SIMD_TRAFO
+static bool check_cpyCoeff( TCoeffOps* ref, TCoeffOps* opt, unsigned num_cases, unsigned W )
+{
+  DimensionGenerator dim;
+
+  static constexpr size_t buf_size = MAX_CU_SIZE * MAX_CU_SIZE;
+
+  // Use xMalloc to create aligned buffers for x86.
+  Pel    *src     = ( Pel*    )xMalloc( Pel,    buf_size );
+  TCoeff *dst_ref = ( TCoeff* )xMalloc( TCoeff, buf_size );
+  TCoeff *dst_opt = ( TCoeff* )xMalloc( TCoeff, buf_size );
+
+  bool passed = true;
+
+  InputGenerator<Pel> inp_gen{ 10, /*is_signed=*/false }; // unsigned 10-bit
```

Comment:
This function is used to copy the prediction residual, thus the value range is **signed** 10 bit (or 8-bit).